tools/kb_generator.py

```diff
@@ -0,0 +1,304 @@
+"""Module responsible for interacting with the OpenAI API to generate KB entries."""
+import dataclasses
+import enum
+import io
+import json
+import os
+import pathlib
+
+import click
+import openai
+import tenacity
+from openai import openai_object
+from openai.api_resources import chat_completion
+
+if os.getenv("DEV") is not None:
```

Comment:
Why the dev env variable?

Reply:
since we don't have `OPENAI_API_KEY` set the unittests will fail, the flag mitigates that